src/plugins/discover/public/application/main/components/top_nav/open_alerts_popover.tsx

```diff
@@ -39,15 +38,13 @@ interface EsQueryAlertMetaData {
 }
 
 export function AlertsPopover({
-  searchSource,
   anchorElement,
-  savedQueryId,
   adHocDataViews,
   services,
+  stateContainer,
   onClose: originalOnClose,
-  updateDataViewList,
 }: AlertsPopoverProps) {
-  const dataView = searchSource.getField('index')!;
+  const dataView = stateContainer.internalState.getState().dataView;
```

Comment:
Alerts flyout seems broken when filters are defined

<img width="500" alt="Screenshot 2023-03-27 at 14 36 31" src="https://user-images.githubusercontent.com/1415710/227950015-cfc30d7f-07b0-44dc-9aac-5cc56999fb9a.png">

Reply:
Thx, I know a quick fix for that, but I won't apply because I need to research, what changed here 

That's the quick fix, seems clone deep in `map_filter.ts` still leaves the filter immutable, which causes the error . But I need to research, why the filter is immutable, not to miss something I'm not aware of

Reply:
Looks good, thanks!